docs(NODE-3569): describe the error API based on instanceof check (#2948)
mongodb/node-mongodb-native: README.md

**documents** collection.

```js
const indexName = await collection.createIndex({ a: 1 });
console.log('index name =', indexName);
```

For more detailed information, see the [indexing strategies page](https://docs.mongodb.com/manual/applications/indexes/).

## Error Handling

If you need to filter certain errors from our driver we have a helpful tree of errors described in [docs/errors.md](https://github.com/mongodb/node-mongodb-native/blob/4.1/docs/errors.md).

It is our recommendation to avoid relying on parsing error message strings in your code, preferring instanceof checks.

> We guarantee the class names of our errors won't change according to semver guidelines, but error messages are subject to change at any time, as well as creating new subclass errors, even patch releases, as we see fit to increase the helpfulness of the errors.

Reviewer: ```suggestion
We guarantee `instanceof` checks will pass according to semver guidelines, but errors may be sub-classed or their messages may change at any time, even patch releases, as we see fit to increase the helpfulness of the errors.
```

Author: accepted 🚀

For more detailed information, see the [indexing strategies page](https://docs.mongodb.com/manual/applications/indexes/).

## Error Handling

If you need to filter certain errors from our driver we have a helpful tree of errors described in [docs/errors.md](https://github.com/mongodb/node-mongodb-native/blob/4.1/docs/errors.md).

> We expect `instanceof` checks to be maintained as our error API under semver as described in the hierarchy errors.md document.

Reviewer 2: The "hierarchy errors.md document" bit reads a little strange to me, could we improve the phrasing there? Also "as our error API" -> "in our error API" ?

> This means we can change the contents of error messages as we see fit to increase the helpfulness of a given message.

Reviewer 2: Just to make it more obvious:
```suggestion
However, we can change the contents of error messages or further subclass errors as we see fit to increase the helpfulness of a given message.
```

> This means we can change the contents of error messages as we see fit to increase the helpfulness of a given message.

Reviewer: I think this bit could be more clear, maybe leading with the first part. Something along the lines of:

"It is our recommendation to avoid relying on parsing error message strings in your code, preferring `instanceof` checks. We guarantee the class names of our errors won't change according to semver guidelines, but error messages are subject to change at any time, even patch releases, as we see fit to increase the helpfulness of a given message."

Reviewer 2: What do you think of my suggestion above regarding subclass phrasing? (since technically we did introduce subclasses outside a major release as the errors work went into a minor version)

> Any new errors we add to the driver will directly extend or extend a child of `MongoError`.

Reviewer 2: ```suggestion
Any new errors we add to the driver will directly extend an existing error class and no existing error will be moved to a different parent class outside of a major release.
```

This means `instanceof` will always be able to capture the errors that our driver throws (or returns in a callback).

```typescript
const client = new MongoClient(url);

> await client.connect()
> const collection = client.db().collection('collection')

Reviewer 2: ```suggestion
await client.connect();
const collection = client.db().collection('collection');
```
